packages/flutter_tools/lib/src/test/web_test_compiler.dart

```diff
@@ -157,7 +201,67 @@ class WebTestCompiler {
     final File manifestFile = outputDirectory.childFile('${output.outputFilename}.json');
     final File sourcemapFile = outputDirectory.childFile('${output.outputFilename}.map');
     final File metadataFile = outputDirectory.childFile('${output.outputFilename}.metadata');
+
     return WebMemoryFS()
       ..write(codeFile, manifestFile, sourcemapFile, metadataFile);
   }
+
+  Future<WebMemoryFS> _compileWasm({
+    required Directory projectDirectory,
+    required String testOutputDir,
+    required List<String> testFiles,
+    required BuildInfo buildInfo,
+    required WebRendererMode webRenderer,
+  }) async {
+    final Directory outputDirectory = _fileSystem.directory(testOutputDir)
+      ..createSync(recursive: true);
+    final File testFile = await _generateTestEntrypoint(
+      testFiles: testFiles,
+      projectDirectory: projectDirectory,
+      outputDirectory: outputDirectory,
+      languageVersion: currentLanguageVersion(_fileSystem, Cache.flutterRoot!),
+    );
+
+    final String dartSdkPath = _artifacts.getArtifactPath(Artifact.engineDartSdkPath, platform: TargetPlatform.web_javascript);
+    final String platformBinariesPath = _artifacts.getHostArtifact(HostArtifact.webPlatformKernelFolder).path;
+    final String platformFilePath = _fileSystem.path.join(platformBinariesPath, 'dart2wasm_platform.dill');
+    final List<String> dartDefines = webRenderer.updateDartDefines(buildInfo.dartDefines);
+    final File outputWasmFile = outputDirectory.childFile('main.dart.wasm');
+
+    final List<String> compilationArgs = <String>[
+      _artifacts.getArtifactPath(Artifact.engineDartBinary, platform: TargetPlatform.web_javascript),
+      'compile',
+      'wasm',
+      '--packages=.dart_tool/package_config.json',
+      '--extra-compiler-option=--dart-sdk=$dartSdkPath',
+      '--extra-compiler-option=--platform=$platformFilePath',
+      '--extra-compiler-option=--multi-root-scheme=org-dartlang-app',
+      '--extra-compiler-option=--multi-root=${projectDirectory.childDirectory('test').path}',
+      '--extra-compiler-option=--multi-root=${outputDirectory.path}',
+      if (webRenderer == WebRendererMode.skwasm) ...<String>[
+        '--extra-compiler-option=--import-shared-memory',
+        '--extra-compiler-option=--shared-memory-max-pages=32768',
+        ],
+      ...buildInfo.extraFrontEndOptions,
+      for (final String dartDefine in dartDefines)
+        '-D$dartDefine',
+
+      '-O1',
+      '-o',
+      outputWasmFile.path,
+      testFile.path, // dartfile
+    ];
+
+    final ProcessUtils processUtils = ProcessUtils(
+      logger: _logger,
+      processManager: _processManager,
+    );
+
+    await processUtils.run(
+      throwOnError: true,
+      compilationArgs,
+    );
+
+    return WebMemoryFS();
```

Comment:
Why are we not populating memory-fs in the wasm case?

Reply:
`WebMemoryFS` is specific to DDC's outputs. To be honest, I'm not entirely sure why we do this at all in the DDC case. In the dart2wasm case, all the compiler outputs are just put into the output directory and they are served directly from there. I didn't want to mess too much with the DDC compilation process in this PR though.

Reply:
Perhaps @jonahwilliams has some insight here

Reply:
writing thousands of 10-100 line files is incredibly slow. So slow that it can easily add an order of magnitude onto the compilation time of the ddc applications. If you're producing a single output file then its irrelevant and I wouldnt use the memory fs